nowcasting_utils/metrics/validation.py

```diff
@@ -27,18 +33,30 @@ def make_validation_results(
     return: Dataframe of predictions and truths
     """
 
-    predictions = pd.DataFrame(
-        predictions, columns=[f"prediction_{i}" for i in range(predictions.shape[1])]
-    )
-    truths = pd.DataFrame(truths, columns=[f"truth_{i}" for i in range(truths.shape[1])])
+    assert predictions_mw.shape == truths_mw.shape
+
+    results_per_forecast_horizon = []
+
+    for i in range(predictions_mw.shape[1]):
```

Comment:
To make it really explicit what the loop is looping round, please consider doing something like:

```python
n_forecast_timesteps = preidctions_mw.shape[1]
for forecast_timestep_i in range(n_forecast_timesteps):
```

(although I'm not sure if `predictions_mw.shape` is the number of timesteps or something else!  So please adapt as appropriate!)

Also:  If this loop is slow, then could this be done in a "vectorised" way (although I'm not entirely sure how!  Maybe using `pd.DataFrame.pivot`?)

Reply:
I did want to do it in a vectorizd way, but I wasnt sure sure how. I'll perhaps leave that one for another time. 

But yea making the loop more clear is a good shout